### ISSUE TITLE
Make compatible with Angular 10

### ISSUE THREAD

**Issue by mruknowme** [MurhafSousli/ngx-progressbar#281]
Updating to Angular 10 leads to the following errors:
```
ERROR in node_modules/@ngx-progressbar/core/lib/ng-progress.module.d.ts:4:50 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

4     static withConfig(config: NgProgressConfig): ModuleWithProviders;
                                                   ~~~~~~~~~~~~~~~~~~~
node_modules/@ng-idle/core/lib/module.d.ts:3:23 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

3     static forRoot(): ModuleWithProviders;
                        ~~~~~~~~~~~~~~~~~~~
node_modules/@ng-idle/keepalive/lib/module.d.ts:3:23 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

3     static forRoot(): ModuleWithProviders;
                        ~~~~~~~~~~~~~~~~~~~
node_modules/@ngx-progressbar/http/lib/ng-progress-http.module.d.ts:4:54 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

4     static withConfig(config: NgProgressHttpConfig): ModuleWithProviders;
                                                       ~~~~~~~~~~~~~~~~~~~
node_modules/@ngx-progressbar/router/lib/ng-progress-router.module.d.ts:6:56 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

6     static withConfig(config: NgProgressRouterConfig): ModuleWithProviders;
```

This happens since ModuleWithProviders has now a required generic type.

Should be easy to update from what I understand: https://angular.io/guide/migration-module-with-providers

Thank you!

**Comment by kamilchlebek** [comment]
@mruknowme Did you use latest version? Seems it was already fixed in `6.0.2`. Commit: https://github.com/MurhafSousli/ngx-progressbar/pull/273/commits/8b1929abe3d091b96e12bc545ac83e01022761cc

**Comment by Karman40** [comment]
same here, but `angular: 11`
